### PR TITLE
add topvar method to get the variable with the same naming

### PR DESCRIPTION
**Pull request by bqpd**
@whoburg, ready for review

(tested in the performance_modeling example)

### CODE REVIEW

**Comment by bqpd** [comment]
test models please

**Comment by bqpd** [comment]
closes #1079

**Comment by bqpd** [comment]
retest this please

**Comment by bqpd** [comment]
passes dependency tests, unit tests are blocked on jenkins right now but it was passing them before so I'm going to merge!

File: gpkit/constraints/set.py [hunk 1]
```diff
@@ -52,28 +52,40 @@ def __getitem__(self, key):
...
-                if arr is not None:
-                    return arr
-            elif len(variables) == 1:
-                return variables[0]
-            raise ValueError("multiple variables are called '%s'; use"
-                             " variables_byname('%s') to see all of them"
-                             % (key, key))
+            return self._choosevar(key, self.variables_byname(key))
+
+    def _choosevar(self, key, variables):
```
Comment (whoburg) [inline gpkit/constraints/set.py]: docstring please?

**Comment by whoburg** [comment]
see also #1094

File: gpkit/constraints/set.py (hunk 1 above)
Comment (whoburg) [inline gpkit/constraints/set.py]: you could push the docstring to the `doctopvar` branch (#1094)

File: gpkit/constraints/set.py [hunk 2]
```diff
@@ -52,28 +52,40 @@ def __getitem__(self, key):
...
+        elif len(variables) == 1:
+            return variables[0]
+        raise ValueError("multiple variables are called '%s'; use"
+                         " variables_byname('%s') to see all of them"
+                         % (key, key))
+
+    def topvar(self, key):
+        "If a variable by a given name exists in the top model, return it"
+        # TODO: could this be done with unique_varkeys?
+        if not self.naming:  # pylint: disable=no-member
```
Comment (whoburg) [inline gpkit/constraints/set.py]: where do `ConstraintSet`s get their `naming` attribute? From `Model` I assume? Shouldn't every `ConstraintSet` have `naming` if we want `topvar` to work?

Comment (bqpd) [inline gpkit/constraints/set.py]: We could give every constraintset a naming, yeah! Right now Models law claim to all variables created in them that aren't in a named submodel, whether those are created in lower-level constraintsets or not, and we could extend that by giving that naming to those constraintsets.

**Pull request merged**